### DISCUSSION TITLE
Reference Error

### DISCUSSION THREAD

**Discussion by marcelosantos32** [community/community#123854] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I'm getting the error 'ReferenceError: variable is not defined' in my JavaScript code. What does this mean and how can I fix it?

**Comment by gabrielbaltar** [comment]
The error message "ReferenceError: variable is not defined" means that you are trying to use a variable in your code that has not been declared or defined yet. This can happen for a few reasons:

Typos: Make sure that you have spelled the variable name correctly both when you declare it and when you use it.
Scope: Variables have a scope, which is the part of your code where they are accessible. If you try to use a variable outside of its scope, you will get this error. To fix this, you need to either move the variable declaration closer to where you are using it, or pass the variable as an argument to a function.
Hoisting: In JavaScript, variable declarations are hoisted to the top of their scope. This means that you can declare a variable after you use it, and it will still be accessible. However, this can be confusing, so it is generally best to declare your variables at the beginning of your code.
